lib/src/fsmonitor.rs

```diff
@@ -26,9 +26,16 @@ use std::str::FromStr;
 /// The recognized kinds of filesystem monitors.
 #[derive(Eq, PartialEq, Clone, Debug)]
 pub enum FsmonitorKind {
-    /// The Watchman filesystem monitor (<https://facebook.github.io/watchman/>).
+    /// Query watchman for changed files.
+    ///
+    /// See the Watchman filesystem monitor (<https://facebook.github.io/watchman/>)
     Watchman,
 
+    /// Registers with watchman to receive notifications of changed files.
+    ///
+    /// See the Watchman filesystem monitor (<https://facebook.github.io/watchman/>)
+    WatchmanTrigger,
```

Comment:
comment: It might be easier to use a configuration option (like `core.fsmonitor.watchman.set_up_trigger`?) rather than an entirely new `FsmonitorKind`, since the normal Watchman code path is essentially the same.

We could communicate the configuration settings via this enum, though. (We might want to rename it if it becomes substantially more than just the "kind" of fsmonitor.)

```suggestion
    /// Query watchman for changed files.
    ///
    /// See the Watchman filesystem monitor (<https://facebook.github.io/watchman/>)
    Watchman {
        set_up_trigger: bool,
    },
```

Reply:
Made this update, but I needed to rework config parsing to give WatchmanKind access to the config object.  Also, I had to use `core.watchman.set_up_trigger`, because `core.fsmonitor.watchman.set_up_trigger` uses `core.fsmonitor` as both a string and a table and `Config` was not happy about that.